components/idp-mgt/org.wso2.carbon.idp.mgt/src/test/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAOTest.java

```diff
@@ -0,0 +1,1486 @@
+    public Object[][] getTotalIdPCountData() {
+
+        ExpressionNode expressionNode1 = new ExpressionNode();
+        List<ExpressionNode> expressionNodesList1 = new ArrayList<>();
+        expressionNodesList1.add(expressionNode1);
+        ExpressionNode expressionNode2 = new ExpressionNode();
+        expressionNode2.setAttributeValue("name");
+        expressionNode2.setOperation("sw");
+        expressionNode2.setValue("test");
+        List<ExpressionNode> expressionNodesList2 = new ArrayList<>();
+        expressionNodesList2.add(expressionNode2);
+
+        return new Object[][]{
+                {SAMPLE_TENANT_ID, expressionNodesList1, 2},
+                {SAMPLE_TENANT_ID, expressionNodesList2, 2},
+                {SAMPLE_TENANT_ID2, expressionNodesList1, 1},
+        };
+    }
+
+    @Test(dataProvider = "getTotalIdPCountData")
+    public void testGetTotalIdPCount(int tenantId, List<ExpressionNode> expressionNodes, int count) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+            int resultCount = cacheBackedIdPMgtDAO.getTotalIdPCount(tenantId, expressionNodes);
+            assertEquals(resultCount, count);
+        }
+    }
+
+    @Test(dataProvider = "getTotalIdPCountData")
+    public void testGetTotalIdPCountException(int tenantId, List<ExpressionNode> expressionNodes, int count)
+            throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+        }
+        assertThrows(IdentityProviderManagementException.class, () ->
+                cacheBackedIdPMgtDAO.getTotalIdPCount(tenantId, expressionNodes));
+    }
+
+    @DataProvider
+    public Object[][] getIdPByNameData() {
+
+        return new Object[][]{
+                {"testIdP1", SAMPLE_TENANT_ID, true},
+                {"testIdP3", SAMPLE_TENANT_ID2, true},
+                {"notExist", SAMPLE_TENANT_ID, false},
+        };
+    }
+
+    @Test(dataProvider = "getIdPByNameData")
+    public void testGetIdPByName(String idpName, int tenantId, boolean isExist) throws Exception {
+
+        mockStatic(IdentityDatabaseUtil.class);
+        try (Connection connection = getConnection(DB_NAME)) {
+            when(IdentityDatabaseUtil.getDBConnection(anyBoolean())).thenReturn(connection);
+            when(IdentityDatabaseUtil.getDataSource()).thenReturn(dataSourceMap.get(DB_NAME));
+            addTestIdps();
+            //retrieving IDP from DB and adding to cache
```

Comment:
Add a space before start the comment, All comments should start with capital and ends with full stop.

Reply:
Check other places as well

Reply:
Resolved with https://github.com/wso2/carbon-identity-framework/pull/4110/commits/28a1194ed9f5de659bd1879f625cd070efdac3fa